doc/releases/changelog-dev.md

```diff
@@ -19,6 +19,9 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The `qml.gradients.hamiltonian_grad` function has been removed.
+  [(#7302)](https://github.com/PennyLaneAI/pennylane/pull/7302)
```

Comment:
```suggestion
* The `qml.gradients.hamiltonian_grad` function has been removed because this gradient recipe is no
  longer required with the :doc:`new operator arithmetic system </news/new_opmath>`.
  [(#7302)](https://github.com/PennyLaneAI/pennylane/pull/7302)
```